### PR TITLE
 Improve test_version_format() error message for version mismatches

### LINKED ISSUE #949: [documentation] library version upgrade fails `test_version_format`
### Feature Request / Improvement

When Pyiceberg upgrades its version, such as now from `0.6.1` -> `0.7.0`. The local repo will fail `test_version_format` until the library is reinstalled with the new version.

Add explanation to [`test_version_format`](https://github.com/apache/iceberg-python/blob/d8bc1ca9af7957ce4d4db99a52c701ac75db7688/tests/test_version.py#L21) error message that the failure can be due to the Pyiceberg library version bump and possible solution such as to rerun `make install`.

```
================================================================== FAILURES ==================================================================
____________________________________________________________ test_version_format _____________________________________________________________

    def test_version_format() -> None:
        from importlib import metadata
    
        installed_version = metadata.version("pyiceberg")
    
>       assert (
            __version__ == installed_version
        ), f"{__version__} <> {installed_version}, the installed version does not match with the current codebase"
E       AssertionError: 0.7.0 <> 0.6.1, the installed version does not match with the current codebase
E       assert '0.7.0' == '0.6.1'
E         - 0.6.1
E         + 0.7.0

tests/test_version.py:26: AssertionError
```

### PR DESCRIPTION
**Pull request by laksh-krishna-sharma**
## What Does This PR Address?

Enhances the test_version_format() Error Message for Version Mismatches.

Fixes #949

## Description

This PR enhances the test_version_format() function in tests/test_version.py by updating the error message to be more informative. The new message provides clearer guidance on resolving version mismatch issues, including a suggestion to rerun make install if needed.

### CODE REVIEW

**Comment by laksh-krishna-sharma** [comment]
Sir I have resolved the issue carefully. Please review the updated.

File: tests/test_version.py [hunk 1]
```diff
@@ -25,4 +25,8 @@ def test_version_format() -> None:
 
     assert (
         __version__ == installed_version
-    ), f"{__version__} <> {installed_version}, the installed version does not match with the current codebase"
+    ), (
+        f"{__version__} <> {installed_version}, the installed version does not match with the current codebase. "
```
Comment (kevinjqliu) [inline tests/test_version.py:29]:
```suggestion
        f"The installed version ({installed_version}) does not match the current codebase version ({__version__})."
```

What do you think of something like this, to make it clear which version is installed and which is current

**Review by kevinjqliu — APPROVED** [review]
LGTM!

r? @sungwy

**Comment by laksh-krishna-sharma** [comment]
Thank you for your feedback, @kevinjqliu . I have resolved the issue carefully . Please review the updated.

**Comment by laksh-krishna-sharma** [comment]
Sir, is there any new or further update?

**Comment by laksh-krishna-sharma** [comment]
I will investigate the issue and address the failing tests as soon as possible. I'll update once the problem is resolved.

**Comment by kevinjqliu** [comment]
Thanks @laksh-krishna-sharma 
Looks like this is a linter issue
```
fix end of files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing tests/test_version.py

```

You can run `make lint` to auto-correct

File: tests/test_version.py [hunk 2]
```diff
@@ -25,4 +25,8 @@ def test_version_format() -> None:
 
     assert (
         __version__ == installed_version
-    ), f"{__version__} <> {installed_version}, the installed version does not match with the current codebase"
+    ), (
+        f"The installed version ({installed_version}) does not match the current codebase version ({__version__})."
+        "This failure could be due to a recent version bump in the Pyiceberg library. "
+        "Please ensure you have the latest version installed by rerunning `make install` command."
+    )
```
Comment (Fokko) [inline tests/test_version.py:32]: We can try to fix it like this:
```suggestion
    )

```

**Review by Fokko — APPROVED** [review]

**Comment by kevinjqliu** [comment]
Closed in favor of #1015

**Pull request closed without merging**